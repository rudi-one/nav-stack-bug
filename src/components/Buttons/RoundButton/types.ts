import {PressableProps} from 'react-native';

export interface RoundButtonProps extends PressableProps {
  text: String;
  size: 'small' | 'big';
}
