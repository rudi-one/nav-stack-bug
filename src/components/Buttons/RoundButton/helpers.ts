import {ViewStyle} from 'react-native';

export const buttonSizeProperties = (size: number): ViewStyle => ({
  width: size,
  height: size,
  borderRadius: size * 0.5,
});
