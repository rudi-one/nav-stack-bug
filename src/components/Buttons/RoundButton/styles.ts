import {StyleSheet} from 'react-native';
import {colors} from 'src/styles/colors';
import {fontFamily, fontSize} from 'src/styles/typography';
import {moderateScaleToMax} from 'src/utils/helpers';
import {buttonSizeProperties} from './helpers';

const bigButtonSize = moderateScaleToMax(150);
const smallButtonSize = moderateScaleToMax(100);

export default StyleSheet.create({
  container: {
    justifyContent: 'center',
    backgroundColor: colors.white,
    overflow: 'hidden',
  },
  pressable: {
    flex: 1,
    justifyContent: 'center',
  },
  bigButton: {
    ...buttonSizeProperties(bigButtonSize),
  },
  smallButton: {
    ...buttonSizeProperties(smallButtonSize),
  },
  text: {
    color: colors.darkBlue,
    textAlign: 'center',
    textTransform: 'uppercase',
    ...fontFamily.coreMellow,
    ...fontSize.x40,
  },
  smallText: {
    ...fontSize.x20,
  },
  disabled: {
    opacity: 0.9,
    backgroundColor: colors.darkMagenta,
  },
});
