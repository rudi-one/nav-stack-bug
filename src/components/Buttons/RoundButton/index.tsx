import React from 'react';
import {Pressable, Text, View} from 'react-native';
import {roundRippleConfig} from 'src/styles/consts';
import styles from './styles';
import {RoundButtonProps} from './types';

const RoundButton: React.FC<RoundButtonProps> = (props: RoundButtonProps) => {
  const {disabled, text, size} = props;
  const isBig = size === 'big';

  return (
    <View
      style={[
        styles.container,
        isBig ? styles.bigButton : styles.smallButton,
        disabled ? styles.disabled : {},
      ]}>
      <Pressable
        {...props}
        android_disableSound={true}
        android_ripple={roundRippleConfig}
        style={styles.pressable}>
        <Text style={[styles.text, !isBig ? styles.smallText : {}]}>
          {text}
        </Text>
      </Pressable>
    </View>
  );
};

export default RoundButton;
