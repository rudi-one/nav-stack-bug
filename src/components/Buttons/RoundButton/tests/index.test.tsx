import {render} from '@testing-library/react-native';
import RoundButton from 'buttons/RoundButton';
import React from 'react';

describe('<Button /> tests', () => {
  it('should match snapshot of small button', () => {
    const component = <RoundButton text={'test'} size={'small'} />;
    expect(render(component)).toMatchSnapshot();
  });
  it('should match snapshot of big button', () => {
    const component = <RoundButton text={'test'} size={'big'} />;
    expect(render(component)).toMatchSnapshot();
  });
  it('should match snapshot of disabled button', () => {
    const component = (
      <RoundButton text={'test'} size={'small'} disabled={true} />
    );
    expect(render(component)).toMatchSnapshot();
  });
});
