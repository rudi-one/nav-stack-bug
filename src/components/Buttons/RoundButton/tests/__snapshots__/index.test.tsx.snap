// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> tests should match snapshot of big button 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "justifyContent": "center",
        "overflow": "hidden",
      },
      Object {
        "borderRadius": 127.5,
        "height": 255,
        "width": 255,
      },
      Object {},
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    size="big"
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
    text="test"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#292C5A",
            "fontFamily": "CoreMellow55Medium",
            "fontSize": 41,
            "textAlign": "center",
            "textTransform": "uppercase",
          },
          Object {},
        ]
      }
    >
      test
    </Text>
  </View>
</View>
`;

exports[`<Button /> tests should match snapshot of disabled button 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "justifyContent": "center",
        "overflow": "hidden",
      },
      Object {
        "borderRadius": 85,
        "height": 170,
        "width": 170,
      },
      Object {
        "backgroundColor": "#a8006c",
        "opacity": 0.9,
      },
    ]
  }
>
  <View
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    size="small"
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
    text="test"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#292C5A",
            "fontFamily": "CoreMellow55Medium",
            "fontSize": 41,
            "textAlign": "center",
            "textTransform": "uppercase",
          },
          Object {
            "fontSize": 27,
          },
        ]
      }
    >
      test
    </Text>
  </View>
</View>
`;

exports[`<Button /> tests should match snapshot of small button 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "justifyContent": "center",
        "overflow": "hidden",
      },
      Object {
        "borderRadius": 85,
        "height": 170,
        "width": 170,
      },
      Object {},
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    size="small"
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
    text="test"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#292C5A",
            "fontFamily": "CoreMellow55Medium",
            "fontSize": 41,
            "textAlign": "center",
            "textTransform": "uppercase",
          },
          Object {
            "fontSize": 27,
          },
        ]
      }
    >
      test
    </Text>
  </View>
</View>
`;
